k-distribution/tests/regression-new/map-symbolic-tests/assignment-3-spec.k.out

```diff
@@ -0,0 +1,4 @@
+<k>
+  assignmentResult ( ( X |-> 1
+  ( MAP [ Y <- 2 ] ) ) [ Z <- 3 ] )
+</k>
```

Comment:
not sure what ` X |-> 1 ( MAP [ Y <- 2 ] )` means.  will it be different from `( X |-> 1 MAP ) [ Y <- 2 ]`?  if so, not sure if it makes any problem.

btw, my expected idea output is `M x |-> 1 y |-> 2 [z <- 3]`.

Reply:
hmm, now i can see that the term is generated by this rule:
```
rule ((K1 |-> V1) MAP) [ K2 <- V2 ] => (K1 |-> V1) (MAP [ K2 <- V2 ]) requires K1 =/=K K2
```

you may need to add more side condition like:
```
rule ((K1 |-> V1) MAP) [ K2 <- V2 ] => (K1 |-> V1) (MAP [ K2 <- V2 ]) requires K1 =/=K K2 andBool notBool K2 in keys(MAP)
```
which will require:
```
rule K1 in (K2 |-> _ MAP) => true when K1 ==K K2
rule K1 in (K2 |-> _ MAP) => K1 in MAP when K1 =/=K K2
```

Reply:
Added tests for `in_keys`